instrumentation/go.mongodb.org/mongo-driver/v2/mongo/otelmongo/mongo.go

```diff
@@ -64,12 +76,57 @@ func (m *monitor) Started(ctx context.Context, evt *event.CommandStartedEvent) {
 	m.Unlock()
 }
 
-func (m *monitor) Succeeded(_ context.Context, evt *event.CommandSucceededEvent) {
+func (m *monitor) Succeeded(ctx context.Context, evt *event.CommandSucceededEvent) {
 	m.Finished(&evt.CommandFinishedEvent, nil)
+
+	hostname, port := peerInfo(evt.ConnectionID)
+	attrs := []attribute.KeyValue{
+		// No need to add semconv.DBSystemMongoDB, it will be added by metrics recorder.
+		semconv.DBOperationName(evt.CommandName),
+		semconv.DBNamespace(evt.DatabaseName),
+		semconv.NetworkPeerAddress(hostname),
+		semconv.NetworkPeerPort(port),
+		semconv.NetworkTransportTCP,
+		semconv.DBResponseStatusCode(statusCodeOK),
+	}
+	// TODO db.query.text attribute is currently disabled by default.
+	// Because event does not provide the query text directly.
+	// command := m.extractCommand(evt)
+	// attrs = append(attrs, semconv.DBQueryText(sanitizeCommand(evt.Command)))
+
+	m.ClientOperationDuration.Record(
+		ctx,
+		evt.Duration.Seconds(),
+		dbconv.SystemNameMongoDB,
+		attrs...,
+	)
 }
 
-func (m *monitor) Failed(_ context.Context, evt *event.CommandFailedEvent) {
+func (m *monitor) Failed(ctx context.Context, evt *event.CommandFailedEvent) {
 	m.Finished(&evt.CommandFinishedEvent, evt.Failure)
+
+	hostname, port := peerInfo(evt.ConnectionID)
+	attrs := []attribute.KeyValue{
+		// No need to add semconv.DBSystemMongoDB, it will be added by metrics recorder.
+		semconv.DBOperationName(evt.CommandName),
+		semconv.DBNamespace(evt.DatabaseName),
+		semconv.NetworkPeerAddress(hostname),
+		semconv.NetworkPeerPort(port),
+		semconv.NetworkTransportTCP,
+		semconv.DBResponseStatusCode(statusCodeError),
```

Comment:
[blocking] The status code should not be static, but reflect [server behavior](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml). Assert the error as [drivers.Error](https://pkg.go.dev/go.mongodb.org/mongo-driver/v2/x/mongo/driver#Error) and pull the code from there.

Reply:
>I've created [GODRIVER-3690](https://jira.mongodb.org/browse/GODRIVER-3690) to provide a stable API as a future improvement.

I'm really happy to this task.

Reply:
@minuk-dev I think it's fine to use the experimental API from the Go Driver, for now. GODRIVER-3690 should be triaged by the drivers team before being worked on, in case there are alternative solutions, etc. For now, I think it's okay to conditionally add the status code:

```go
if code, ok := evt.Failure.(driver.Error); ok {
	attrs = append(attrs, semconv.DBResponseStatusCode(strconv.Itoa(code)))
}
```

CC: @matthewdale